Cannot start language server:  Message: URI "" is not valid

### Server Version
<!--
Run `terraform-ls --version` to show the version, and paste the result between the ``` marks below.
If you are not running the latest version, please try upgrading because your issue may have already been fixed.
-->
```
0.3.2
```

### Terraform Version
<!--
Run `terraform -v to show the version, and paste the result between the ``` marks below.
Make sure you are running the same binary that the language server would normally pick up from $PATH
if you have more than one version installed on your system
-->
```
Terraform v0.12.26
```

### Client Version
<!--
Please share what IDE and/or plugin which interacts with the server
e.g. Sublime Text (LSP plugin) v0.9.7
-->
```
VSCode 1.46.0
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file.
For security, you can also encrypt the files using HashiCorp's GPG public key published at
https://www.hashicorp.com/security#secure-communications
-->


```hcl
provider "azurerm" {
  version = "~> 2.0"
  features { }
}
```

### Log Output
<!--
Full debug output can be obtained by launching server with particular flag (e.g. -log-file)
Please follow instructions in docs/TROUBLESHOOTING.md

Please create a GitHub Gist containing the debug output. Please do *NOT* paste
the debug output in the issue, since it may be long.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using HashiCorp's GPG public key published at
https://www.hashicorp.com/security#secure-communications
-->

```text
Launching language server: c:\Users\OisinGrehan\.vscode\extensions\hashicorp.terraform-2.0.1/lsp/terraform-ls serve
2020/06/11 15:59:16 service.go:63: Preparing new session ...
2020/06/11 15:59:16 langserver.go:79: Starting server (pid 14976) ...
2020/06/11 15:59:16 server.go:469: Received 1 new requests
2020/06/11 15:59:16 server.go:165: Processing 1 requests
2020/06/11 15:59:16 server.go:223: Checking request for "initialize": {"processId":38236,"rootPath":null,"rootUri":null,"capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true,"resourceOperations":["create","rename","delete"],"failureHandling":"textOnlyTransactional"},"didChangeConfiguration":{"dynamicRegistration":true},"didChangeWatchedFiles":{"dynamicRegistration":true},"symbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]}},"executeCommand":{"dynamicRegistration":true},"configuration":true,"workspaceFolders":true},"textDocument":{"publishDiagnostics":{"relatedInformation":true},"synchronization":{"dynamicRegistration":true,"willSave":true,"willSaveWaitUntil":true,"didSave":true},"completion":{"dynamicRegistration":true,"contextSupport":true,"completionItem":{"snippetSupport":true,"commitCharactersSupport":true,"documentationFormat":["markdown","plaintext"],"deprecatedSupport":true,"preselectSupport":true},"completionItemKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}},"hover":{"dynamicRegistration":true,"contentFormat":["markdown","plaintext"]},"signatureHelp":{"dynamicRegistration":true,"signatureInformation":{"documentationFormat":["markdown","plaintext"],"parameterInformation":{"labelOffsetSupport":true}}},"definition":{"dynamicRegistration":true,"linkSupport":true},"references":{"dynamicRegistration":true},"documentHighlight":{"dynamicRegistration":true},"documentSymbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"hierarchicalDocumentSymbolSupport":true},"codeAction":{"dynamicRegistration":true,"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"codeLens":{"dynamicRegistration":true},"formatting":{"dynamicRegistration":true},"rangeFormatting":{"dynamicRegistration":true},"onTypeFormatting":{"dynamicRegistration":true},"rename":{"dynamicRegistration":true,"prepareSupport":true},"documentLink":{"dynamicRegistration":true},"typeDefinition":{"dynamicRegistration":true,"linkSupport":true},"implementation":{"dynamicRegistration":true,"linkSupport":true},"colorProvider":{"dynamicRegistration":true},"foldingRange":{"dynamicRegistration":true,"rangeLimit":5000,"lineFoldingOnly":true},"declaration":{"dynamicRegistration":true,"linkSupport":true}}},"trace":"off","workspaceFolders":null}
2020/06/11 15:59:16 rpc_logger.go:29: Incoming request for "initialize" (ID 0): {"processId":38236,"rootPath":null,"rootUri":null,"capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true,"resourceOperations":["create","rename","delete"],"failureHandling":"textOnlyTransactional"},"didChangeConfiguration":{"dynamicRegistration":true},"didChangeWatchedFiles":{"dynamicRegistration":true},"symbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]}},"executeCommand":{"dynamicRegistration":true},"configuration":true,"workspaceFolders":true},"textDocument":{"publishDiagnostics":{"relatedInformation":true},"synchronization":{"dynamicRegistration":true,"willSave":true,"willSaveWaitUntil":true,"didSave":true},"completion":{"dynamicRegistration":true,"contextSupport":true,"completionItem":{"snippetSupport":true,"commitCharactersSupport":true,"documentationFormat":["markdown","plaintext"],"deprecatedSupport":true,"preselectSupport":true},"completionItemKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}},"hover":{"dynamicRegistration":true,"contentFormat":["markdown","plaintext"]},"signatureHelp":{"dynamicRegistration":true,"signatureInformation":{"documentationFormat":["markdown","plaintext"],"parameterInformation":{"labelOffsetSupport":true}}},"definition":{"dynamicRegistration":true,"linkSupport":true},"references":{"dynamicRegistration":true},"documentHighlight":{"dynamicRegistration":true},"documentSymbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"hierarchicalDocumentSymbolSupport":true},"codeAction":{"dynamicRegistration":true,"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"codeLens":{"dynamicRegistration":true},"formatting":{"dynamicRegistration":true},"rangeFormatting":{"dynamicRegistration":true},"onTypeFormatting":{"dynamicRegistration":true},"rename":{"dynamicRegistration":true,"prepareSupport":true},"documentLink":{"dynamicRegistration":true},"typeDefinition":{"dynamicRegistration":true,"linkSupport":true},"implementation":{"dynamicRegistration":true,"linkSupport":true},"colorProvider":{"dynamicRegistration":true},"foldingRange":{"dynamicRegistration":true,"rangeLimit":5000,"lineFoldingOnly":true},"declaration":{"dynamicRegistration":true,"linkSupport":true}}},"trace":"off","workspaceFolders":null}
2020/06/11 15:59:16 rpc_logger.go:45: Error for "initialize" (ID 0): [-32098] URI "" is not valid
2020/06/11 15:59:16 server.go:204: Completed 1 requests [14.354ms elapsed]
[Error - 3:59:16 p.m.] Server initialization failed.
  Message: URI "" is not valid
  Code: -32098 
```

### Expected Behavior
<!-- What should have happened? -->
Language server should start.
### Actual Behavior
<!-- What actually happened? -->
Language server fails to start.
### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. Open a folder in IDE XYZ
2. Open file example.tf from that folder
3. Trigger autocompletion on line 5, column 1 (1-indexed)
-->
f1, "enable terraform language serve"

--- a/langserver/handlers/initialize.go
+++ b/langserver/handlers/initialize.go
@@ -28,13 +28,12 @@ func (lh *logHandler) Initialize(ctx context.Context, params lsp.InitializeParam
 	}
 
 	fh := ilsp.FileHandlerFromDirURI(params.RootURI)
-	if !fh.Valid() {
-		return serverCaps, fmt.Errorf("URI %q is not valid", params.RootURI)
-	}
-
-	if !fh.IsDir() {
+	if fh.URI() == "" || !fh.IsDir() {
 		return serverCaps, fmt.Errorf("Editing a single file is not yet supported." +
 			" Please open a directory.")
+	}
+	if !fh.Valid() {
+		return serverCaps, fmt.Errorf("URI %q is not valid", params.RootURI)
 	}
 
 	err := lsctx.SetClientCapabilities(ctx, &params.Capabilities)

